### PR TITLE
Peter zhizhin: Added an option for PPO to train on incomplete episode

### PR DESCRIPTION
**Pull request by tfboyd**
Taken from this PR that got stale.  https://github.com/tensorflow/agents/pull/36

I just merged Peter Zhinzhin's changes and fixed the conflicts.  There is no attempt to take any credit. Peter's commits are in this PR with my final commit doing the merge conflict.  

Getting ready to run unit tests, will update this commit when done.

### CODE REVIEW

**Comment by tfboyd** [comment]
Marking CLA yes as Peter as signed it in the other PR and this is me merging everything.  This is on the up and up.

**Comment by tfboyd** [comment]
Trying my kokoro unit tests.  :-)

**Comment by tfboyd** [comment]
@PeterZhizhin  Just heads up I made a new PR with your commits.

File: tf_agents/agents/ppo/ppo_utils.py [hunk 1]
```diff
@@ -24,29 +24,36 @@
...
+    episode_is_complete = tf.cumsum(
+        tf.cast(batched_next_time_step.is_last(), tf.float32),
+        axis=1,
+        reverse=True) > 0
 
   # 1.0 for all valid timesteps. 0.0 where between episodes.
   not_between_episodes = ~batched_next_time_step.is_first()
-
-  return tf.cast(episode_is_complete & not_between_episodes, tf.float32)
+  if episode_is_complete is None:
```
Comment (sguada) [inline tf_agents/agents/ppo/ppo_utils.py:53]:
```
mask = not_between_episodes
if allow_partial_episodes:
  mask = mask & episode_is_complete
return tf.cast(mask, tf.float32)
```

File: tf_agents/agents/ppo/ppo_utils.py [hunk 2]
```diff
@@ -24,29 +24,36 @@
...
   """
-  # 1.0 for timesteps of all complete episodes. 0.0 for incomplete episode at
-  #   the end of the sequence.
-  episode_is_complete = tf.cumsum(
-      tf.cast(batched_next_time_step.is_last(), tf.float32),
-      axis=1,
-      reverse=True) > 0
+  if allow_partial_episodes:
+    episode_is_complete = None
+  else:
```
Comment (sguada) [inline tf_agents/agents/ppo/ppo_utils.py:43]: `if not allow_partial_episodes:`

**Review by sguada — CHANGES_REQUESTED** [review]

File: tf_agents/agents/ppo/ppo_agent.py [hunk 3]
```diff
@@ -165,6 +166,9 @@ def __init__(self,
       reward_norm_clipping: Value above and below to clip normalized reward.
       normalize_observations: If true, keeps moving mean and variance of
         observations and normalizes incoming observations.
+      train_on_partial_episodes: If True (default False), then data from
```
Comment (ebrevdo) [inline tf_agents/agents/ppo/ppo_agent.py:169]: let's remove this arg and assume it's always True; per oscar's comment on original PR.

**Comment by tfboyd** [comment]
@kuanghuei

**Pull request merged**